src/components/webview/WebviewPanel.ts

```diff
@@ -0,0 +1,120 @@
+import { Disposable, Uri, ViewColumn, WebviewPanel, window } from 'vscode';
+import { WebviewResolver } from './WebviewResolver';
+import { WebviewMessage } from './webviewEvents';
+import { Message, MessageBus, MessageKind } from '../messageBus';
+
+type PanelOptions = Readonly<{
+	type: string;
+	title: string;
+	showOptions:
```

Comment:
IMHO the fewer possibilities of states, the better.

I would not have `showOptions` attribute at all, but a `viewColumn` attribute and `preserveFocus: boolean`